### DISCUSSION TITLE
Customization of a CommonToolbarItem?

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#4428] [Q&A] [answered]
The following code snippet from my UIProvider adds a CommonToolbarItem with the icon "icon-help".

1. Is there a way to change either the foreground or background color of this button?
2. Is there a way to configure CommonToolbarItem to use other images besides the ones delivered with iTwin?

```
public provideToolbarButtonItems(_stageId: string, stageUsage: string, toolbarUsage: ToolbarUsage, toolbarOrientation: ToolbarOrientation) : CommonToolbarItem[]  {
    const toolbarButtonItems: CommonToolbarItem[] = [];
    if ((toolbarUsage === ToolbarUsage.ViewNavigation) && (toolbarOrientation === ToolbarOrientation.Vertical)) {
      toolbarButtonItems.push ( ToolbarItemUtilities.createActionButton("About", 50, "icon-help", "About Traffic Design", (): void => { 
        ModalDialogManager.openDialog(createAboutDialog());
      }));  
```

Thanks

**Comment by NancyMcCallB** [comment, marked as answer]
The colors of the components are determined by the color theme in AppUi.
To use your own image, you can import an svg directly 

```
import myIconSvg from "./myicon.svg"; // or whatever the path

const customActionButton = ToolbarItemUtilities.createActionButton(
        "custom-action-button",
        -1,
        IconSpecUtilities.createWebComponentIconSpec(myIconSvg ),
        "Custom Action Button",
        (): void => {
          // your code
        },
      );
```
Or if you have a React icon, you can do something like:

```
    ToolbarHelper.createToolbarItemFromItemDef(145, new CommandItemDef({
      commandId: "My Command",
      iconSpec: <ReactIcon />,
      label: "My Command",
      execute: () => {
                // your code
      },
    }), { groupPriority: 30 }),
```
When you just pass in the icon name as you have above, you are limited to the icons delivered in the web font. We'll likely deprecate the web font soon, but you can find all those icons as React icons in the itwinui-react-icons package.

Let me know if you run into problems!

Nancy

**Reply by JohnMajerle** [reply]
Hey Nancy,

Worked the first time!  See image snippet below.  Thanks for the assist.

![image](https://user-images.githubusercontent.com/30668739/194076909-e6c35b3c-e555-46b4-9fed-5702cde2e4ec.png)
